mantidproject/mantid
# Added LoadSampleEnvironment Algorithm

**Description of work.**
Added `LoadSampleEnvironment` algorithm that works similarly to `LoadSampleShape`, and made changes to `LoadStl` and `meshObject` to allow them to support much larger files which are likely to be used for environments.

**To test:**
create a sample workspace, set the log level to debug, then run `LoadSampleEnvironment` with add set to false, the input and output workspaces set to you sample workspace, and the Filename to `/ExternalData/Testing/Data/UnitTest/cubeBin.stl` (you'll need to build StandardTestData) and everything else its default. the following should be in the log
```
Environment has: 1 elements.
Environment Material: 
Environment Material Number Density: 0.000000
```
preceded by a list of groups of 3 double's, these are the vertices of the environments meshobject.

next run `LoadSampleEnvironment` again, with the same workspaces but add set to true, the log should be the same except for `Environment has: 2 elements.` from here on continue running with add set to true, using the vertices to check that the translation and rotation matrices work properly for valid matrices. 

Finally check that setting `setMaterial` to true unlocks the remaining options in the dialog, these options should follow the same rules of allowed inputs as `SetSampleMaterial`, turn add off, to overwrite the environment (resetting back to `Environment has: 1 elements.`) as the outputs
```
Environment Material: 
Environment Material Number Density:
```
only show those values for the container, and not any additional parts of the environment, setting the chemical formula should effect the `Environment Material`, whereas the sample Density options should effect the `Environment Material Number Density`, (it should be equal to sampleNumberDensity if that is entered, otherwise it will be calculated based off the other options)

Fixes #23807
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
